src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.Windows.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Runtime.InteropServices
+{
+    public partial class MarshalTests
+    {
+        public static IEnumerable<object[]> IsComImport_Windows_ReturnsExpected()
+        {
+            yield return new object[] { new ComImportObject(), true };
+            yield return new object[] { new SubComImportObject(), true };
+            yield return new object[] { new InterfaceAndComImportObject(), true };
+            yield return new object[] { new InterfaceComImportObject(), false };
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
```

Comment:
The reason why this test fails is because the comimport guid is not present on Nano. I borrowed it from the System.DirectoryServices source code.

I couldn't find any other guids that worked on classes as opposed to interfaces, which is what this PR is supposed to support.